### PR TITLE
Fixes #120 Migrated to new plugin lib go

### LINKED ISSUE #120: Update to use new plugin lib go to avoid importing Snap
When running `make` the `vendor` directory is created and the Snap directory is downloaded. We need to use the new plugin lib so Snap is not imported.

### PR DESCRIPTION
**Pull request by kdembler**
Fixes #120 

Summary of changes:
- Migrated plugin to use new plugin lib
- Updated tests

### CODE REVIEW

File: .sync.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@
   sudo: true
```
Comment (katarzyna-z) [inline .sync.yml]: Consider making changes connected with `snap-pluginsync` and configuration for it in separated pull request, see also list of managed files in .pluginsync.yml

File: influxdb/influxdb.go [hunk 2]
```diff
@@ -33,16 +31,13 @@ import (
...
-	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
 )
 
 const (
-	name       = "influxdb"
-	version    = 21
-	pluginType = plugin.PublisherPluginType
+	Name       = "influxdb"
+	Version    = 21
```
Comment (katarzyna-z) [inline influxdb/influxdb.go]: Please increment plugin version

File: influxdb/influxdb.go [hunk 3]
```diff
@@ -189,19 +123,38 @@ func (f *influxPublisher) Publish(contentType string, content []byte, config map
 	}
 
 	//Set up batch points
+	database, err := config.GetString("database")
```
Comment (katarzyna-z) [inline influxdb/influxdb.go]: I suggest moving parts for checking configurable parameters into separated function, this comment is also for other parts of code where configurable parameters are checked.

File: influxdb/influxdb.go [hunk 4]
```diff
@@ -212,38 +165,39 @@ func (f *influxPublisher) Publish(contentType string, content []byte, config map
...
+		if _, ok := m.Tags["unit"]; !ok {
+			tags["unit"] = m.Unit
 		}
 
 		// Process the tags for this metric
-		for k, v := range m.Tags() {
+		for k, v := range m.Tags {
 			// Convert the standard tag describing where the plugin is running to "source"
-			if k == core.STD_TAG_PLUGIN_RUNNING_ON {
+
```
Comment (katarzyna-z) [inline influxdb/influxdb.go]: Please remove this empty line

File: influxdb/influxdb.go [hunk 5]
```diff
@@ -540,18 +484,18 @@ func groupCommonNamespaces(m plugin.MetricType, tags map[string]string, mpoints
 	mkeys = append(mkeys, s2l.Strings()...)
 
 	// Converts the map keys to a string key
-	sk := strings.Join(mkeys, core.Separator)
+	sk := strings.Join(mkeys, "\U0001f422")
```
Comment (katarzyna-z) [inline influxdb/influxdb.go]: I suggest making const for `"\U0001f422"` with describing name

File: influxdb/influxdb_medium_test.go [hunk 6]
```diff
@@ -64,97 +60,125 @@ func TestInfluxPublish(t *testing.T) {
...
 	Convey(" Publish integer metric via "+scheme, func() {
-		metrics := []plugin.MetricType{
-			*plugin.NewMetricType(core.NewNamespace("foo"), time.Now(), tags, "some unit", 99),
+		metrics := []plugin.Metric{
+			{
+				Namespace: plugin.NewNamespace("foo"),
+				Timestamp: time.Now(),
+				Config:    mcfg,
+				Tags:      tags,
+				Unit:      "integer",
```
Comment (katarzyna-z) [inline influxdb/influxdb_medium_test.go]: I've noticed that in test after your changes there is correlation between `Data` and `Unit` but user cannot expect that in unit there will be information about data type of metric, as unit I understand for example bytes. Consider changing  into previous version.

File: influxdb/influxdb_medium_test.go [hunk 7]
```diff
@@ -18,20 +18,17 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
 	"net/http"
 	"os"
 	"strings"
 	"testing"
 	"time"
 
-	"github.com/intelsdi-x/snap/control/plugin"
-	"github.com/intelsdi-x/snap/core"
-	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
```
Comment (katarzyna-z) [inline influxdb/influxdb_medium_test.go]: please remove this empty line to have consistency between medium and small tests

File: main.go [hunk 8]
```diff
@@ -20,13 +20,11 @@ limitations under the License.
 package main
 
 import (
-	"os"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
```
Comment (katarzyna-z) [inline main.go]: please remove this empty line ;)

File: .sync.yml (hunk 1 above)
Comment (kdembler) [inline .sync.yml]: Changes were required in this pull request for it to not fail Travis build.

File: influxdb/influxdb.go [hunk 9]
```diff
@@ -67,93 +63,86 @@ func init() {
...
-	config.Add(r10)
-
-	r11, err := cpolicy.NewStringRule("scheme", false, HTTP)
-	handleErr(err)
-	r11.Description = "Influxdb communication protocol"
-	config.Add(r11)
-
-	cp.Add([]string{""}, config)
-	return cp, nil
+func (ip *InfluxPublisher) getConfig(config plugin.Config) error {
```
Comment (katarzyna-z) [inline influxdb/influxdb.go]: It seems that this part of code in not safe when two tasks will get the same instance of the plugin, the config map should be protected.

Comment (kdembler) [inline influxdb/influxdb.go]: Thanks for that, fixed! :smile:

File: influxdb/influxdb.go [hunk 10]
```diff
@@ -212,38 +187,38 @@ func (f *influxPublisher) Publish(contentType string, content []byte, config map
...
 			tags[k] = v
 		}
 
-		data := m.Data()
+		data := m.Data
 
 		//publishing of nil value causes errors
 		if data == nil {
-			log.Errorf("Received nil value of metric, this metric is not published, namespace: %s, timestamp: %s", m.Namespace().String(), m.Timestamp().String())
+			log.Errorf("Received nil value of metric, this metric will not be published, namespace: %s, timestamp: %s", m.Namespace.Strings()[0], m.Timestamp.String())
```
Comment (lmroz) [inline influxdb/influxdb.go]: `m.Namespace.Strings()[0]` will give you first element of namespace while original code was giving whole namespace. If library is not providing method to join elements print it as a value `"%v"`

File: influxdb/influxdb.go [hunk 11]
```diff
@@ -67,93 +63,82 @@ func init() {
...
-	config.Add(r10)
-
-	r11, err := cpolicy.NewStringRule("scheme", false, HTTP)
-	handleErr(err)
-	r11.Description = "Influxdb communication protocol"
-	config.Add(r11)
-
-	cp.Add([]string{""}, config)
-	return cp, nil
+func getConfig(config plugin.Config) (map[string]interface{}, error) {
```
Comment (lmroz) [inline influxdb/influxdb.go]: Perhaps this code would look better if configuration was read to structure with typed fields.

File: influxdb/influxdb.go [hunk 12]
```diff
@@ -33,17 +31,15 @@ import (
...
 )
 
 const (
-	name       = "influxdb"
-	version    = 21
-	pluginType = plugin.PublisherPluginType
+	Name       = "influxdb"
+	Version    = 22
+	PluginType = "publisher"
 	maxInt64   = ^uint64(0) / 2
```
Comment (lmroz) [inline influxdb/influxdb.go]: Side note, there is `math.MaxInt64` for such constants :face_with_head_bandage:

File: influxdb/influxdb.go (hunk 11 above)
Comment (kdembler) [inline influxdb/influxdb.go]: You're right, asserting type with every read from config doesn't look very pretty, fixed now.

File: influxdb/influxdb.go (hunk 10 above)
Comment (kdembler) [inline influxdb/influxdb.go]: Missed that one, fixed now.

**Pull request merged**